### PR TITLE
Updater for LineRider

### PR DESCRIPTION
**Pull request by Sussy-OS**
Is *this* how you make an updater script? If I messed something up, please tell me.

### CODE REVIEW

**Comment by Sussy-OS** [comment]
@Botspot Can you please check if I did this correctly?

**Comment by Sussy-OS** [comment]
Or, anybody who has done this before.

**Comment by cycool29** [comment]
Instead of using releases to bump the version number, I think you can create a repo, use a GitHub Action to release LineRider periodically, and use `get_release` to get the latest release tag in the update script.

My [whatsapp-for-linux](https://github.com/cycool29/whatsapp-for-linux) repo is a good example of that.

**Comment by Sussy-OS** [comment]
> Instead of using releases to bump the version number, I think you can create a repo, use a GitHub Action to release LineRider periodically, and use `get_release` to get the latest release tag in the update script.
> 
> My [whatsapp-for-linux](https://github.com/cycool29/whatsapp-for-linux) repo is a good example of that.

@cycool29 Okay, would you like to help? (I can do it myself, I'm just asking.)

**Comment by cycool29** [comment]
How did you build LineRider?

**Comment by Sussy-OS** [comment]
can we make it, so that it will build, after I make a commit, at 12:00 PM EST?

**Comment by cycool29** [comment]
> > How did you build LineRider?
> 
> @cycool29 [This link, the BUILD part](https://github.com/Sussy-OS/LRA-Community-Edition#build) Please use my fork, instead of RatharBeLunar' as mine fixes a bug on linux/rpi.

Can I build it on Linux?
I can't build it on my Raspberry Pi, the `nuget` command failed, and Visual Studio is only available on Windows. 



> can we make it, so that it will build, after I make a commit, at 12:00 PM EST?

You can set the time when the workflow will run in your workflow file.

**Comment by Sussy-OS** [comment]
> > > How did you build LineRider?
> > 
> > 
> > @cycool29 [This link, the BUILD part](https://github.com/Sussy-OS/LRA-Community-Edition#build) Please use my fork, instead of RatharBeLunar' as mine fixes a bug on linux/rpi.
> 
> Can I build it on Linux? I can't build it on my Raspberry Pi, the `nuget` command failed, and Visual Studio is only available on Windows.
> 
> > can we make it, so that it will build, after I make a commit, at 12:00 PM EST?
> 
> You can set the time when the workflow will run in your workflow file.

No you cant build it on linux. And I only want it to build, when I make a commit on the repo. (The one that I sent how to build.)

**Comment by cycool29** [comment]
> And I only want it to build, when I make a commit on the repo. 

Yes, that is possible. Just set `on: push` in your workflow file, so it will run every time you commit a change.

> No you cant build it on linux.

I can't help you much if LineRider needs to be built on Windows, since I am not familiar with Windows commands.

**Comment by Sussy-OS** [comment]
> > No you cant build it on linux.
> 
> I can't help you much if LineRider needs to be built on Windows, since I am not familiar with Windows commands.

Oh okay. Why do you need to build it?

**Comment by cycool29** [comment]
> Instead of using releases to bump the version number, I think you can create a repo, use a GitHub Action to build LineRider periodically and release it, and use `get_release` to get the latest release tag in the update script.

^^

**Comment by Sussy-OS** [comment]
> > Instead of using releases to bump the version number, I think you can create a repo, use a GitHub Action to build LineRider periodically and release it, and use `get_release` to get the latest release tag in the update script.
> 
> ^^

@cycool29 Is it possible to build using visual studio on gh actions?

**Comment by cycool29** [comment]
> @cycool29 Is it possible to build using visual studio on gh actions?

Not the GUI app (Visual Studio), but the CLI tool `msbuild`.

https://github.com/marketplace/actions/setup-msbuild

**Comment by Sussy-OS** [comment]
> > @cycool29 Is it possible to build using visual studio on gh actions?
> 
> Not the GUI app (Visual Studio), but the CLI tool `msbuild`.
> 
> https://github.com/marketplace/actions/setup-msbuild

Okay, thanks. I will start to work on the gh actions tool.

**Comment by Sussy-OS** [comment]
@cycool29 Once it makes the build, where does it save it?

**Comment by cycool29** [comment]
> @cycool29 Once msbuild makes the build, where is it saved?

Not sure about that. I have not used `msbuild` before.

**Comment by Sussy-OS** [comment]
@Botspot , @cycool29 I am ready for merge, I tested the script. It works fine. Please merge.

File: .github/workflows/updates/LineRider.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+webVer=$(get_release Sussy-OS/LRA-Community-Edition)
+all_url="https://github.com/Sussy-OS/LRA-Community-Edition/releases/download/${webVer}/LineRider.game.auto-release.${webVer}-LR.zip"
+pi_apps_ver="$(cat 'install' | grep 'version=' | sed 's/version=//')"
```
Comment (cycool29) [inline .github/workflows/updates/LineRider.sh:5]:
```suggestion
```

You don't need to specify `pi_apps_ver` in the script. The `update_github_script.sh` will do that for you.

**Comment by cycool29** [comment]
@Sussy-OS Why are you putting zip in a zip?

Also, I am not a collaborator of pi-apps, I can't merge this for you, lol.

File: .github/workflows/updates/LineRider.sh (hunk 1 above)
Comment (Sussy-OS) [reply .github/workflows/updates/LineRider.sh]: thx

**Comment by Sussy-OS** [comment]
> @Sussy-OS Why are you putting zip in a zip?
> 
> Also, I am not a collaborator of pi-apps, I can't merge this for you, lol.

It's how github artifacts does it, though I am going to try to fix it. Okay, I will just wait for @Botspot

**Comment by Sussy-OS** [comment]
@cycool29 There, now it works. I had to remove & edit some code in the workflow, on my [LineRider](https://github.com/Sussy-OS/LRA-Community-Edition) repo.

**Comment by Sussy-OS** [comment]
Still waiting for Botspot to merge

**Review by cycool29 — APPROVED** [review]
Looks nice to me. @Botspot

**Pull request merged**